Src/Coravel/Scheduling/Schedule/Event/ScheduledEvent.cs

```diff
@@ -48,6 +48,13 @@ public ScheduledEvent(Func<Task> scheduledAsyncTask)
             return scheduledEvent;
         }
 
+        internal static ScheduledEvent WithInvocableAndParams(Type invocableType, IServiceScopeFactory scopeFactory, object[ ] parameters)
+        {
+            var scheduledEvent = WithInvocableType(scopeFactory, invocableType);
```

Comment:
In the method above that this new one is based on (e.g. `WithInvocableAndParams<T>()`), there is a condition that the type `T` must implement `IInvocable`. This new method doesn't check to make sure the Type passed in as a parameter meets that condition.

Also, just to note, we want to do this here and not in the `Scheduler` version of this method since this is done at a lower level and will apply this restriction to any callers too (even though there's only one right now).

Could you add a guard clause at the beginning of this method to:
a. Test that the type implements `Invocable` (keep in mind the scenario where `Type` inherits from another class which implements `IInvocable` too 😉).
b. If not met, then throw an exception with the message: `When using ScheduleWithParams<T>() you must supply a Type that inherits from IInvocable.`

Please add tests around this exception too 👍

Reply:
I have added the guard clause as requested and I have also added a test to verify the type check.

Reply:
I overlooked this in the initial PR. I have added the guard code now along with the test case to validate it.